### DISCUSSION TITLE
unit test status

### DISCUSSION THREAD

**Discussion by themighty1** [evilsocket/opensnitch#525] [Development]
@gustavo-iniguez-goya , I noticed some commits in your repo with tests, however you did not merge them yet into this branch.
Can we have a separate branch in this repo where we could work on adding tests?
If you havent already, I would like to write some integration tests while opensnitch is running:
-start an app, add a rule for it, try to send some data, check if the data got through/was blocked.

**Comment by gustavo-iniguez-goya** [comment]
Yes, I've written some tests. I was about to commit them to the master branch here.

If we add them on a separate branch, can we integrate them with Actions?

**Comment by themighty1** [comment]
Nice. OK, let's just go ahead and merge them into master.

**Comment by gustavo-iniguez-goya** [comment]
added

**Comment by themighty1** [comment]
Thanks, could you also elaborate on what command to use to run those tests?

**Comment by gustavo-iniguez-goya** [comment]
yes, from the `rule/` or `procmon/` directories, run: `go test -v`

**Comment by gustavo-iniguez-goya** [comment]
It would be good to have the Discussions section enabled, to discuss these things and other ideas.

**Comment by themighty1** [comment]
OK, this thread could be moved to discussions.

**Comment by gustavo-iniguez-goya** [comment]
I've started adding GUI tests: https://github.com/evilsocket/opensnitch/commit/ff99b39d091fb743675d75f9991b336c8587aa99

I'll keep adding more in the coming days. Also it seems that they could be integrated with github actions: https://codito.in/xvfb-in-github-action/
